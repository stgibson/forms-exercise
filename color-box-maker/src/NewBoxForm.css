.NewBoxForm {
  margin-bottom: 20px;
}