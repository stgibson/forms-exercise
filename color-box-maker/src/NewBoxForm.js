import React, { useState } from "react";
import "./NewBoxForm.css";

const NewBoxForm = ({ addBox }) => {
  const initFormData = { width: "", height: "", color: "" };
  const [formData, setFormData] = useState(initFormData);

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData(() => ({ ...formData, [name]: value }));
  };

  const handleSubmit = (event) => {
    event.preventDefault();
    addBox(formData.width, formData.height, formData.color);
    setFormData(initFormData);
  };

  return (
    <form className="NewBoxForm" onSubmit={ handleSubmit }>
      <label htmlFor="width">Width:</label>
      <input
        id="width"
        name="width"
        type="number"
        min="1"
        value={ formData.width }
        onChange={ handleChange }
      />
      <label htmlFor="height">Height:</label>
      <input
        id="height"
        name="height"
        type="number"
        min="1"
        value={ formData.height }
        onChange={ handleChange }
      />
      <label htmlFor="color">Color:</label>
      <input
        id="color"
        name="color"
        type="text"
        min="1"
        value={ formData.color }
        onChange={ handleChange }
      />
      <button type="submit">Add Box</button>
    </form>
  );
};

export default NewBoxForm;