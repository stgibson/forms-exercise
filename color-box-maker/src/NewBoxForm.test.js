import React from "react";
import NewBoxForm from "./NewBoxForm";
import { render } from "@testing-library/react";

it("renders without crashing", () => {
  render(<NewBoxForm />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<NewBoxForm />);
  expect(asFragment()).toMatchSnapshot();
});