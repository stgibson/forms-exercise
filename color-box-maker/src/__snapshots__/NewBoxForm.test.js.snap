// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="NewBoxForm"
  >
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      min="1"
      name="width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      min="1"
      name="height"
      type="number"
      value=""
    />
    <label
      for="color"
    >
      Color:
    </label>
    <input
      id="color"
      min="1"
      name="color"
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Box
    </button>
  </form>
</DocumentFragment>
`;
